src/components/organisms/EarthEditor/TagPane/LayerTagPane/hooks.ts

```diff
@@ -0,0 +1,77 @@
+import { useCallback } from "react";
+
+import useCommonHooks from "../commonHooks";
+
+export default () => {
+  const {
+    loading,
+    attachTagGroupToLayer,
+    attachTagItemToLayer,
+    createTagGroup,
+    createTagItem,
+    sceneTagGroups,
+    updateTagGroup,
+    layerTagGroups,
+    detachTagGroupFromLayer,
+    detachTagItemFromLayer,
+  } = useCommonHooks();
+
+  const handleCreateTagGroup = useCallback(
+    async (label: string) => {
+      const tagGroup = await createTagGroup(label);
+      if (!tagGroup?.data?.createTagGroup?.tag.id) return;
+      await attachTagGroupToLayer(tagGroup?.data?.createTagGroup?.tag.id);
+    },
+    [attachTagGroupToLayer, createTagGroup],
+  );
+
+  const handleCreateTagItem = useCallback(
+    async (label: string, tagGroupId: string) => {
+      const tagItem = await createTagItem(label, tagGroupId);
+      if (!tagItem?.data?.createTagItem?.tag.id) return;
+      await attachTagItemToLayer(tagItem?.data?.createTagItem?.tag.id);
+    },
+    [attachTagItemToLayer, createTagItem],
+  );
+
+  const handleAttachTagGroupToLayer = useCallback(
+    async (tagGroupId: string) => {
+      await attachTagGroupToLayer(tagGroupId);
+    },
+    [attachTagGroupToLayer],
+  );
+
+  const handleAttachTagItemToLayer = useCallback(
+    async (tagItemId: string) => {
+      await attachTagItemToLayer(tagItemId);
+    },
+    [attachTagItemToLayer],
+  );
+
+  const handleDetachTagGroupFromLayer = useCallback(
+    async (tagGroupId: string) => {
+      await detachTagGroupFromLayer(tagGroupId);
+    },
+    [detachTagGroupFromLayer],
+  );
+
+  const handleDetachTagItemFromLayer = useCallback(
+    async (tagId: string) => {
+      await detachTagItemFromLayer(tagId);
+    },
+    [detachTagItemFromLayer],
+  );
```

Comment:
Same as I said for the sceneTagPane as well. I think you are repeating lots. And again, if this is part of the refactor you'll do later, you can just resolve this comment for now.